channels/reducer.js

```diff
@@ -41,31 +42,38 @@ function createReducer (config) {
       }
     }),
     [unsetChannels]: (state, action) => loop(state, Effects.batch(
-      map(action.payload, channel => Effects.constant(unsetChannel(channel))))),
+      map(action.payload, id => Effects.constant(unsetChannel(id))))),
     [unsetChannel]: (state, action) => {
-      const channel = action.payload
-      const { id, channels, clips } = channel
+      const channel = state.records[action.payload]
+      const { id, channelIds = [], clipIds = [] } = channel
 
-      const nextRecords = state.records
-      delete nextRecords[id]
+      // remove this channel from all parent channels
+      // TODO: is this necessary? what if we just assume an empty id doesnt exist?
```

Comment:
wdyt? last we talked, you said it was the responsibility of the item being deleted to make sure all references to it are removed. this seems like it breaks encapsulation - in this case it works because channels are nested, but how can deleteSample know which clips to remove itself from? given especially the sample reducer does not have access to the clips store